gen3/bin/kube-setup-metadata-delete-expired-objects-cronjob.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Deploy the `metadata-delete-expired-objects` cronjob.
+#
+
+source "${GEN3_HOME}/gen3/lib/utils.sh"
+gen3_load "gen3/gen3setup"
+
+setup_config() {
+  gen3_log_info "check metadata-delete-expired-objects secret"
+  local secretsFolder="$(gen3_secrets_folder)/g3auto/metadata-delete-expired-objects"
+  if [[ ! -f "$secretsFolder/config.json" ]]; then
+    local hostname=$(gen3 api hostname)
+    gen3_log_info "kube-setup-metadata-delete-expired-objects-job" "creating fence oidc client for $hostname"
+    local secrets=$(g3kubectl exec -c fence $(gen3 pod fence) -- fence-create client-create --client metadata-delete-expired-objects-job --grant-types client_credentials | tail -1)
+    # secrets looks like ('CLIENT_ID', 'CLIENT_SECRET')
+    if [[ ! $secrets =~ (\'(.*)\', \'(.*)\') ]]; then
+        # try delete client
+        g3kubectl exec -c fence $(gen3 pod fence) -- fence-create client-delete --client metadata-delete-expired-objects-job > /dev/null 2>&1
```

Comment:
I didn't completely understand what we are trying to achieve here. I see we were trying to execute `fence-create` to create a client, but then if the secret is not in the format expected, it looks like we just delete the client and recreate again. Something doesn't seem right to me. Are you sure there is no better way than this? Or am I missing something altogether?

Reply:
this is a bit of a pattern in cloud-automation scripts - we try to create the client, and if the output is not in the expected format `(client id, client secret)`, we assume it's because the command is returning an error because the client already exists. The fact that we're in this block in the first place means that the config file does not exist, so the creds for this old client have been lost. So we try deleting the old client before creating it. If that doesn't work, then an existing client is not the problem, and we return an error